File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/fragment/BaseFragment.java
package com.blogspot.e_kanivets.moneytracker.fragment;

import android.support.annotation.LayoutRes;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.blogspot.e_kanivets.moneytracker.R;

/**
 * Base (abstract) class to encapsulate some common fragment operations.
 * Created on 3/12/16.
 *
 * @author Evgenii Kanivets
 */
public abstract class BaseFragment extends Fragment {
    @SuppressWarnings("unused")
    private static final String TAG = "BaseFragment";

    protected abstract void initViews(View rootView);

    protected void inflateAppBarLayout(@LayoutRes int layoutResId) {
        ViewGroup container = (ViewGroup) getActivity().findViewById(R.id.app_bar_container);
        if (container == null) return;

        container.removeAllViews();

        if (layoutResId == -1) return;

        LayoutInflater.from(getActivity()).inflate(layoutResId, container, true);
    }
}

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/base/BaseFragment.kt
package com.blogspot.e_kanivets.moneytracker.activity.base

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.blogspot.e_kanivets.moneytracker.MtApp
import com.blogspot.e_kanivets.moneytracker.di.AppComponent

abstract class BaseFragment : Fragment() {

    protected val appComponent: AppComponent = MtApp.get().appComponent

    protected abstract val contentViewId: Int

    protected abstract fun initData()

    protected abstract fun initViews(view: View)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initData()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
        inflater.inflate(contentViewId, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initViews(view)
    }
}
